spec/models/address_spec.rb

```diff
@@ -253,5 +253,115 @@ def addr_details(addr, expected_msg)
 
     end
 
+
+    describe '#self.geocode_all_needed(sleep_between: 0.5, num_per_batch: 50)' do
+
+      let(:a_company) { create(:company, num_addresses: 0) }
+
+      let(:norbotten_region) { create(:region, name: 'Norrbotten') }
+      let(:overtornea_kommun) { create(:kommun, name: 'Övertorneå') }
+
+      # These are real addresses in  Övertorneå Municipality in Norrbotten County:
+
+      let(:valid_address1) { addr1 = create(:address,
+                                    street_address: 'Matarengivägen 24',
+                                    post_code: '957 31',
+                                    city: 'Övertorneå',
+                                    kommun: overtornea_kommun,
+                                    region: norbotten_region,
+                                    addressable: a_company )
+                            addr1.validate
+                            addr1
+       }
+
+      let(:valid_address2) { addr2 = create(:address,
+                                    street_address: 'Skolvägen 12',
+                                    post_code: '957 31',
+                                    city: 'Övertorneå',
+                                    kommun: overtornea_kommun,
+                                    region: norbotten_region,
+                                    addressable: a_company )
+                            addr2.validate
+                            addr2
+       }
+
+      let(:valid_address3) { addr3 = create(:address,
+                                            street_address: 'Matarengivägen 30',
+                                            post_code: '957 31',
+                                            city: 'Övertorneå',
+                                            kommun: overtornea_kommun,
+                                            region: norbotten_region,
+                                            addressable: a_company )
+                              addr3.validate
+                              addr3
+       }
+
+      it 'nothing geocoded if all have latitude and longitude' do
+        valid_address1
+        valid_address2
+        valid_address3
+
+        need_geocoding = Address.not_geocoded
+        needed_geocoding = need_geocoding.count
+
+        Address.geocode_all_needed
+
+        after_run_need_geocoding = Address.not_geocoded.count
+
+        expect(needed_geocoding).to eq 0
+        expect(after_run_need_geocoding).to eq 0
+      end
+
+
+      it 'will geocode 1 that needs it' do
+
+        valid_address1
+        valid_address2
+        valid_address3
+
+        query = <<-SQL
+          UPDATE addresses SET latitude=NULL, longitude=NULL  
+           WHERE street_address = 'Matarengivägen 24'
+        SQL
```

Comment:
Curious as to why go direct to sql?  Doe the geocoder gem protect against setting these attributes to `nil` via AR methods?